src/Clapi/Relay.hs

```diff
@@ -166,3 +170,18 @@ handleUnclaimedMessages vs msgs = if isValidClaim then handleOwnerMessages vs ms
             _ -> "Top level path already claimed"
         _ -> "Bundle claims multiple toplevel paths"
     errorAll = map (\m -> MsgError (msgPath' m) "Not within known toplevel path") msgs
+
+handleClientMessages :: Valuespace Validated -> [Message] -> [Message]
+handleClientMessages vs msgs = rmsgs ++ subMsgs
+  where
+    subMsgs = concatMap createMsgs subPaths
+    createMsgs p = [assignMsg p] ++ nodeMsgs p
+    assignMsg p = case getType vs p of
+        (Just tp) -> MsgAssignType p tp
```

Comment:
Danger, incomplete pattern match!
(I realise that this shouldn't fail given that we have been given a validated Valuespace)